Getting int-short assignment error

Hi, after command `make` getting the next error
`OBJECTS +++ include/drat/string/j.c +++ out/include/drat/string/j.o
gcc -std=c99 -D _GNU_SOURCE -Werror -Wall -Wextra -Wno-incompatible-pointer-types -Wno-multichar -Wno-unused-variable -Wno-unused-parameter -Wno-missing-field-initializers -I./include -I/usr/local/Cellar/argp-standalone/1.3/include/ -c "include/drat/string/j.c" -o "out/include/drat/string/j.o"
include/drat/string/j.c:53:55: error: format specifies type 'unsigned short' but
      the argument has type 'int' [-Werror,-Wformat]
  ...extended fields:  %"PRIu16"\n", has_xfields ? xfields->xf_num_exts : 0);
                       ~~~~~~~~      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [out/include/drat/string/j.o] Error 1`

Why it could be?
-GNU C Compiler (gcc) 
-GNU Make (make)
-GNU Argp library (<argp.h>)
are installed. I have MacOs 10.15.7

> Out of interest, to see where this incompatibility arises, what is your build environment (OS, hardware, etc.)?

Thanks for the reply! So I use MacOs 10.15.7 6-Core Intel Core i5 (3.7), 40Gb RAM, if it is really matter.
A pulled port-test, run it again, but still have an exception
```
gcc -std=c99 -D _GNU_SOURCE -Werror -Wall -Wextra -Wno-incompatible-pointer-types -Wno-multichar -Wno-unused-variable -Wno-unused-parameter -Wno-missing-field-initializers -I./include -I/usr/local/Cellar/argp-standalone/1.3/include/ -c "include/drat/string/j.c" -o "out/include/drat/string/j.o"
include/drat/string/j.c:54:55: error: format specifies type 'unsigned short' but
      the argument has type 'int' [-Werror,-Wformat]
  ...fields:  %"PRIu16"\n", has_xfields ? xfields->xf_num_exts : UINT16_C(0));
              ~~~~~~~~      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [out/include/drat/string/j.o] Error 1
➜  drat git:(port-test) 
```
Hope did everything right

--- a/include/drat/string/j.c
+++ b/include/drat/string/j.c
@@ -11,6 +11,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdint.h>
 
 #if defined(__APPLE__) || defined(__BSD__)
     #include <sys/stat.h>
@@ -50,7 +51,7 @@
 
 // Helper function for printing objects which have xfields
 static void print_xf_details(bool has_xfields, xf_blob_t* xfields) {
-    printf("Number of extended fields:  %"PRIu16"\n", has_xfields ? xfields->xf_num_exts : 0);
+    printf("Number of extended fields:  %"PRIu16"\n", has_xfields ? xfields->xf_num_exts : UINT16_C(0));
     if (has_xfields) {
         printf("Details of extended fields:\n\n");
         xf_pair_t** xf_pairs_array = get_xf_pairs_array(xfields);

--- a/include/drat/string/j.c
+++ b/include/drat/string/j.c
@@ -51,7 +51,7 @@
 
 // Helper function for printing objects which have xfields
 static void print_xf_details(bool has_xfields, xf_blob_t* xfields) {
-    printf("Number of extended fields:  %"PRIu16"\n", has_xfields ? xfields->xf_num_exts : UINT16_C(0));
+    printf("Number of extended fields:  %"PRIu16"\n", (uint16_t)(has_xfields ? xfields->xf_num_exts : 0));
     if (has_xfields) {
         printf("Details of extended fields:\n\n");
         xf_pair_t** xf_pairs_array = get_xf_pairs_array(xfields);

